framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -767,6 +767,38 @@ public Range refineEqualTo(Range right) {
     }
 
     /**
+     * Refines this range to reflect that no value in it can be equal to a value in the given range.
+     * This is used for calculating the control-flow-refined result of the != operator. For example:
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 15) int b;
+     *     ...
+     *     if (a != b) {
+     *         // range of <i>a</i> is now refined to [0, 2] because a value in range [3, 10]
```

Comment:
Ah, my mistake. It's `!=` that doesn't provide too much information. And actually I should have leave a comment in the `Range` class for future reference (like in you case..) 

Thanks for working on this. It's kind of a remaining work of my part. Other than this issue, this PR looks great to me.